### ISSUE TITLE
Remote GM

### ISSUE THREAD

**Issue by k9lego** [daid/EmptyEpsilon#543]
How hard would it be to program a remote GM console for headless servers?

**Comment by daid** [comment]
The GM station uses information and functions that aren't replicated across the network. Which is why it's server only, it's not designed like that, so it would be a lot of work.

**Comment by amir-arad** [comment]
what's the scope of the difference?
say, what GM function would be trivial to do remotely? say I was thinking about quickly adding co-GM stations, what could co-GMs do?

**Comment by daid** [comment]
My guess, only the GM comms functions would work right now. Enemy AI, tweaking objects, creating objects, all done server side only. Viewing the whole "world" on a client can be done, sans AI information.

**Comment by tdelc** [comment]
I think that a co-GM station, even only for the communication would be a
very great thing.

**Comment by k9lego** [comment]
I agree with that. GM would be more effective if one person made the story and one managed the enemies.

**Comment by amir-arad** [comment]
I'd postpone the security thing, it's not a blocker IMO. as it is, players either get auto-connecting stations, or are free to choose, so what's the harm of them choosing a co-GM screen instead of any other "wrong" screen?

**Comment by amir-arad** [comment]
re-visiting this discussion, as i expect to need co-GM stations in the foreseeable future.
I would appreciate a review on this idea:
expressing GM actions in  LUA (very small scripts) so that the local GM screen will run them locally, and a remote screen will send them to the server using a single designated GM_ACTION multiplayer command.
putting execution aside, is that approach practical? is there expected to be an issue with drag-and-drop?
